app/javascript/actions/bots.js

```diff
@@ -8,40 +8,45 @@ export const RECEIVE = 'BOTS_RECEIVE'
 export const RECEIVE_ERROR = 'BOTS_RECEIVE_ERROR'
 export const CREATE_SUCCESS = 'BOTS_CREATE_SUCCESS'
 
+export const botsFetch = () : T.BotsAction => ({
+  type: FETCH
+})
+
+export const botsReceive = (items : T.ById<T.Bot>) : T.BotsAction => ({
+  type: RECEIVE,
+  items
+})
+
+export const botsReceiveError = () : T.BotsAction => ({
+  type: RECEIVE_ERROR
+})
+
+export const botsCreateSuccess = (bot : T.Bot) : T.BotsAction => ({
+  type: CREATE_SUCCESS,
+  bot
+})
+
 export const fetchBots = () => (dispatch : T.Dispatch, getState : T.GetState) => {
   const state = getState()
 
   // Don't fetch Bots if they are already being fetched
-  if ((state.bots.fetching : boolean)) {
+  if (state.bots.fetching) {
```

Comment:
👎  unless `( ... : boolean)` is used it wont detect typos/errors in outer property in guards due to undefined and falsey semantics of javascript.

Alternative `( ... : *)` could be used to force some type.

Ref: https://github.com/facebook/flow/issues/3887

Reply:
Ugh... I should have asked why you were making the explicit cast.